### DISCUSSION TITLE
help run Blockscout

### DISCUSSION THREAD

**Discussion by Ranofty** [0xPolygon/polygon-edge#783] [General] [answered]
Hello please i need help to run blockscout am already done with all the processes but can't open it on browser
below are my logs 

 sudo systemctl start explorer.service

sudo systemctl status explorer.service

● explorer.service - chainscan Server
     Loaded: loaded (/etc/systemd/system/explorer.service; enabled; vendor pres>
     Active: activating (auto-restart) (Result: exit-code) since Sun 2022-10-09>
    Process: 89264 ExecStart=/usr/local/bin/mix phx.server (code=exited, status>
   Main PID: 89264 (code=exited, status=1/FAILURE)

Oct 09 09:09:14 ip-172-31-68-184 systemd[1]: explorer.service: Failed with resu>
lines 1-7/7 (END)

sudo journalctl -u explorer.service -f

45: Config.__import__!/1
Oct 09 09:11:45 ip-172-31-68-184 mix[92502]:     (stdlib 3.17.1) erl_eval.erl:685: :erl_eval.do_apply/6
Oct 09 09:11:45 ip-172-31-68-184 mix[92502]:     (stdlib 3.17.1) erl_eval.erl:123: :erl_eval.exprs/5
Oct 09 09:11:45 ip-172-31-68-184 mix[92502]:     (elixir 1.13.4) lib/code.ex:404: Code.validated_eval_string/3
Oct 09 09:11:45 ip-172-31-68-184 mix[92502]:     (stdlib 3.17.1) erl_eval.erl:685: :erl_eval.do_apply/6
Oct 09 09:11:45 ip-172-31-68-184 systemd[1]: explorer.service: Main process exited, code=exited, status=1/FAILURE
Oct 09 09:11:45 ip-172-31-68-184 systemd[1]: explorer.service: Failed with result 'exit-code'.
Oct 09 09:11:46 ip-172-31-68-184 systemd[1]: explorer.service: Scheduled restart job, restart counter is at 1416.
Oct 09 09:11:46 ip-172-31-68-184 systemd[1]: Stopped chainscan Server.
Oct 09 09:11:46 ip-172-31-68-184 systemd[1]: Started chainscan Server.
Oct 09 09:11:47 ip-172-31-68-184 mix[92540]: ** (File.Error) could not read file "/usr/local/blockscout/apps/block_scout_web/config/prod# run in production mode.exs": no such file or directory
Oct 09 09:11:47 ip-172-31-68-184 mix[92540]:     (elixir 1.13.4) lib/file.ex:355: File.read!/1
Oct 09 09:11:47 ip-172-31-68-184 mix[92540]:     (elixir 1.13.4) lib/config.ex:245: Config.__import__!/1
Oct 09 09:11:47 ip-172-31-68-184 mix[92540]:     (stdlib 3.17.1) erl_eval.erl:685: :erl_eval.do_apply/6
Oct 09 09:11:47 ip-172-31-68-184 mix[92540]:     (stdlib 3.17.1) erl_eval.erl:123: :erl_eval.exprs/5
Oct 09 09:11:47 ip-172-31-68-184 mix[92540]:     (elixir 1.13.4) lib/code.ex:404: Code.validated_eval_string/3
Oct 09 09:11:47 ip-172-31-68-184 mix[92540]:     (stdlib 3.17.1) erl_eval.erl:685: :erl_eval.do_apply/6
Oct 09 09:11:47 ip-172-31-68-184 systemd[1]: explorer.service: Main process exited, code=exited, status=1/FAILURE
Oct 09 09:11:47 ip-172-31-68-184 systemd[1]: explorer.service: Failed with result 'exit-code'.
Oct 09 09:11:48 ip-172-31-68-184 systemd[1]: explorer.service: Scheduled restart job, restart counter is at 1417.
Oct 09 09:11:48 ip-172-31-68-184 systemd[1]: Stopped chainscan Server.
Oct 09 09:11:48 ip-172-31-68-184 systemd[1]: Started chainscan Server.
Oct 09 09:11:49 ip-172-31-68-184 mix[92578]: ** (File.Error) could not read file "/usr/local/blockscout/apps/block_scout_web/config/prod# run in production mode.exs": no such file or directory
Oct 09 09:11:49 ip-172-31-68-184 mix[92578]:     (elixir 1.13.4) lib/file.ex:355: File.read!/1
Oct 09 09:11:49 ip-172-31-68-184 mix[92578]:     (elixir 1.13.4) lib/config.ex:245: Config.__import__!/1
Oct 09 09:11:49 ip-172-31-68-184 mix[92578]:     (stdlib 3.17.1) erl_eval.erl:685: :erl_eval.do_apply/6
Oct 09 09:11:49 ip-172-31-68-184 mix[92578]:     (stdlib 3.17.1) erl_eval.erl:123: :erl_eval.exprs/5
Oct 09 09:11:49 ip-172-31-68-184 mix[92578]:     (elixir 1.13.4) lib/code.ex:404: Code.validated_eval_string/3
Oct 09 09:11:49 ip-172-31-68-184 mix[92578]:     (stdlib 3.17.1) erl_eval.erl:685: :erl_eval.do_apply/6
Oct 09 09:11:49 ip-172-31-68-184 systemd[1]: explorer.service: Main process exited, code=exited, status=1/FAILURE
Oct 09 09:11:49 ip-172-31-68-184 systemd[1]: explorer.service: Failed with result 'exit-code'.
Oct 09 09:11:50 ip-172-31-68-184 systemd[1]: explorer.service: Scheduled restart job, restart counter is at 1418.
Oct 09 09:11:50 ip-172-31-68-184 systemd[1]: Stopped chainscan Server.
Oct 09 09:11:50 ip-172-31-68-184 systemd[1]: Started chainscan Server.
Oct 09 09:11:50 ip-172-31-68-184 mix[92616]: ** (File.Error) could not read file "/usr/local/blockscout/apps/block_scout_web/config/prod# run in production mode.exs": no such file or directory
Oct 09 09:11:50 ip-172-31-68-184 mix[92616]:     (elixir 1.13.4) lib/file.ex:355: File.read!/1
Oct 09 09:11:50 ip-172-31-68-184 mix[92616]:     (elixir 1.13.4) lib/config.ex:245: Config.__import__!/1
Oct 09 09:11:50 ip-172-31-68-184 mix[92616]:     (stdlib 3.17.1) erl_eval.erl:685: :erl_eval.do_apply/6
Oct 09 09:11:50 ip-172-31-68-184 mix[92616]:     (stdlib 3.17.1) erl_eval.erl:123: :erl_eval.exprs/5
Oct 09 09:11:50 ip-172-31-68-184 mix[92616]:     (elixir 1.13.4) lib/code.ex:404: Code.validated_eval_string/3
Oct 09 09:11:50 ip-172-31-68-184 mix[92616]:     (stdlib 3.17.1) erl_eval.erl:685: :erl_eval.do_apply/6
Oct 09 09:11:50 ip-172-31-68-184 systemd[1]: explorer.service: Main process exited, code=exited, status=1/FAILURE
Oct 09 09:11:50 ip-172-31-68-184 systemd[1]: explorer.service: Failed with result 'exit-code'.
Oct 09 09:11:52 ip-172-31-68-184 systemd[1]: explorer.service: Scheduled restart job, restart counter is at 1419.
Oct 09 09:11:52 ip-172-31-68-184 systemd[1]: Stopped chainscan Server.
Oct 09 09:11:52 ip-172-31-68-184 systemd[1]: Started chainscan Server.
Oct 09 09:11:52 ip-172-31-68-184 mix[92654]: ** (File.Error) could not read file "/usr/local/blockscout/apps/block_scout_web/config/prod# run in production mode.exs": no such file or directory
Oct 09 09:11:52 ip-172-31-68-184 mix[92654]:     (elixir 1.13.4) lib/file.ex:355: File.read!/1
Oct 09 09:11:52 ip-172-31-68-184 mix[92654]:     (elixir 1.13.4) lib/config.ex:245: Config.__import__!/1
Oct 09 09:11:52 ip-172-31-68-184 mix[92654]:     (stdlib 3.17.1) erl_eval

sudo netstat -tulpn

Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      2869/systemd-resolv
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      794/sshd: /usr/sbin
tcp6       0      0 :::22                   :::*                    LISTEN      794/sshd: /usr/sbin
udp        0      0 0.0.0.0:37999           0.0.0.0:*                           22146/avahi-daemon:
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           22146/avahi-daemon:
udp        0      0 127.0.0.53:53           0.0.0.0:*                           2869/systemd-resolv
udp        0      0 172.35.62.187:69        0.0.0.0:*                           2863/systemd-networ
udp6       0      0 :::37399                :::*                                22146/avahi-daemon:
udp6       0      0 :::5353                 :::*                                22146/avahi-daemon:

i need your help on this please

**Comment by laviniat1996** [comment]
Hi @Ranofty,

Thank you for opening this discussion. 
Firstly, I would like to ask you if you checked out our [Blockscout guide](https://wiki.polygon.technology/docs/edge/additional-features/blockscout)?

**Reply by Ranofty** [reply]
Yes sure that’s the exactly Guide I follow

**Reply by Ranofty** [reply]
Sometimes it will show the log below but it will still not work in the browser

 sudo systemctl status explorer.service

● explorer.service - chainscan Server
     Loaded: loaded (/etc/systemd/system/explorer.service; enabled; vendor pres>
     Active: active (running) since Mon 2022-10-10 07:17:03 UTC; 2ms ago
   Main PID: 1813766 (elixir)
      Tasks: 1 (limit: 18962)
     Memory: 128.0K
     CGroup: /system.slice/explorer.service
             ├─1813766 /bin/sh /usr/local/bin/elixir /usr/local/bin/mix phx.ser>
             ├─1813767 /bin/sh /usr/local/bin/elixir /usr/local/bin/mix phx.ser>
             └─1813768 [dirname]

Oct 10 07:17:03 ip-172-31-68-184 systemd[1]: Started chainscan Server.
lines 1-12/12 (END)

**Reply by ZeljkoBenovic** [reply]
@Ranofty Have you enabled [Production Mode](https://wiki.polygon.technology/docs/edge/additional-features/blockscout/#set-production-mode) for Elixir framework before starting Blockscout compilation?
 
I'd suggest, deleting cloned blockscout folder, and starting again from [Part 3](https://wiki.polygon.technology/docs/edge/additional-features/blockscout/#part-3---clone-and-compile-blockscout) but keep in mind that `MIX_ENV` needs to be set to `prod` before blockscout compilation process is started.

**Reply by Ranofty** [reply]
Yes sure i did that in all the processes also i follow all the steps there but am suspecting my error in 3 places

1, When running `createuser --interactive`  i got this error

createuser --interactive
Enter name of role to add: blockscout
Shall the new role be a superuser? (y/n) y
createuser: error: creation of new role failed: ERROR:  must be superuser to create superusers

2, when `sudo touch /etc/systemd/system/explorer.service` and `sudo vi /etc/systemd/system/explorer.service`

[Unit]
Description=chainscan Server
After=network.target
StartLimitIntervalSec=0

[Service]
Type=simple
Restart=always
RestartSec=1
User=root
StandardOutput=syslog
StandardError=syslog
WorkingDirectory=/usr/local/blockscout
ExecStart=/usr/local/bin/mix phx.server
EnvironmentFile=/usr/local/blockscout/env_vars.env

[Install]
WantedBy=multi-user.target

i don't really know exactly what to replace there as am new to Elixir 

3,  This `cd apps/block_scout_web
mix phx.gen.cert blockscout blockscout.local` i always run this command which am cofused on exactly how to use it

Am waiting for your respond

**Reply by laviniat1996** [reply]
@Ranofty Unfortunately, we are not able to assist with this anymore, because it is more of a devops issue, not a Polygon Edge one. Also, Blockscout is not our own production, just an integration, so we cannot provide full support for it. But, if you still need help, we can guide you to the consultance team. 

Thank you for your understanding.

**Comment by Ranofty** [comment, marked as answer]
Thanks for your reply Guiding to the consultance team will really help

also what is polygon edge `ETHEREUM_JSONRPC_VARIANT` ? I need to know this so that I can go ahead and use blockscout guide to install 

I know polygon edge is not using Geth then which one is it using? 

erigon, geth, nethermind, besu? Or which one?

**Reply by laviniat1996** [reply]
Hi @Ranofty , 

If you'd like to have consultance, please let me know, but I want to let you know that is is paid. 
To answer you question, we've made it work with `nethermind` variant.

**Reply by Ranofty** [reply]
Thanks now working 🤩🙏

**Reply by laviniat1996** [reply]
@Ranofty We are glad to heart that ^_^
